File: 05_arrays/05_04_remove_duplicates_from_sorted_array/RemoveDuplicatesFromSortedArray.java
public class RemoveDuplicatesFromSortedArray {
    public static void main(String[] args) {
        int[] nums = new int[] {0, 0, 1, 1, 1, 2, 2, 3, 3, 4};
        System.out.println("Elements:-");
        for (int i : nums) {
            System.out.print(i + " ");
        }
        System.out.println();
        int nonDuplicates = removeDuplicates(nums);
        System.out.println("Non-duplicates: " + nonDuplicates);
        System.out.println("Elements:-");
        for (int i = 0; i < nonDuplicates; i++) {
            System.out.print(i + " ");
        }
    }

    private static int removeDuplicates(int[] nums) {
        int j = 1;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] != nums[i-1]) {
                nums[j++] = nums[i];
            }
        }

        return j;
    }
}

File: 05_arrays/05_04_remove_duplicates_from_sorted_array/remove_duplicates_from_sorted_array.cpp
#include <bits/stdc++.h>

using namespace std;

void print(vector<int>& nums) {
    for (size_t i = 0; i < nums.size(); i++)
    {
        cout << nums[i] << " ";
    }
}

int removeDuplicates(vector<int>& nums) {
    int j = 1;
    for (size_t i = 1; i < nums.size(); i++)
    {
        if (nums[i] != nums[i-1])
        {
            nums[j++] = nums[i];
        }
    }
    return j;
}

int main() {
    vector<int> nums = {0, 0, 1, 1, 1, 2, 2, 3, 3, 4};

    cout << "Elements:-" << endl;
    print(nums);
    int nonDuplicates = removeDuplicates(nums);
    cout << endl;
    cout << "No. of non duplicates: " << nonDuplicates << endl;
    
    for (size_t i = 0; i < nonDuplicates; i++)
    {
        cout << nums[i] << " ";
    }
    
}